tests/model_registry/rbac/conftest.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+import shlex
+import subprocess
+import os
+from typing import Generator, List, Dict, Any
+from ocp_resources.namespace import Namespace
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.role_binding import RoleBinding
+from ocp_resources.role import Role
+from kubernetes.dynamic import DynamicClient
+from pyhelper_utils.shell import run_command
+from tests.model_registry.utils import generate_random_name, generate_namespace_name
+from simple_logger.logger import get_logger
+from tests.model_registry.constants import MR_INSTANCE_NAME
+
+
+LOGGER = get_logger(name=__name__)
+DEFAULT_TOKEN_DURATION = "10m"
+
+
+@pytest.fixture(scope="function")
+def sa_namespace(request: pytest.FixtureRequest, admin_client: DynamicClient) -> Generator[Namespace, None, None]:
+    """
+    Creates a temporary namespace using a context manager for automatic cleanup.
+    Function scope ensures a fresh namespace for each test needing it.
+    """
+    test_file = os.path.relpath(request.fspath.strpath, start=os.path.dirname(__file__))
+    ns_name = generate_namespace_name(file_path=test_file)
+    LOGGER.info(f"Creating temporary namespace: {ns_name}")
+    with Namespace(client=admin_client, name=ns_name) as ns:
+        ns.wait_for_status(status=Namespace.Status.ACTIVE, timeout=120)
+        LOGGER.info(f"Namespace {ns_name} is active.")
+        yield ns
+        LOGGER.info(f"Namespace {ns_name} deletion initiated by context manager.")
+
+
+@pytest.fixture(scope="function")
+def service_account(admin_client: DynamicClient, sa_namespace: Namespace) -> Generator[ServiceAccount, None, None]:
+    """
+    Creates a ServiceAccount within the temporary namespace using a context manager.
+    Function scope ensures it's tied to the lifetime of sa_namespace for that test.
+    """
+    sa_name = generate_random_name(prefix="mr-test-user")
+    LOGGER.info(f"Creating ServiceAccount: {sa_name} in namespace {sa_namespace.name}")
+    with ServiceAccount(client=admin_client, name=sa_name, namespace=sa_namespace.name, wait_for_resource=True) as sa:
+        LOGGER.info(f"ServiceAccount {sa_name} created.")
+        yield sa
+        LOGGER.info(f"ServiceAccount {sa_name} deletion initiated by context manager.")
+
+
+@pytest.fixture(scope="function")
+def sa_token(service_account: ServiceAccount) -> str:
+    """
+    Retrieves a short-lived token for the ServiceAccount using 'oc create token'.
+    Function scope because token is temporary and tied to the SA for that test.
+    """
+    sa_name = service_account.name
+    namespace = service_account.namespace
+    LOGGER.info(f"Retrieving token for ServiceAccount: {sa_name} in namespace {namespace}")
+    try:
+        cmd = f"oc create token {sa_name} -n {namespace} --duration={DEFAULT_TOKEN_DURATION}"
+        LOGGER.debug(f"Executing command: {cmd}")
+        res, out, err = run_command(command=shlex.split(cmd), verify_stderr=False, check=True, timeout=30)
```

Comment:
I think we need to use hide_log_command option here: https://github.com/RedHatQE/pyhelper-utils/blob/main/pyhelper_utils/shell.py#L23. We don't want to display the token information in the console. Can you please double check?

Reply:
I can also confirm it does not print the token
```
2025-05-07T12:46:40.366859 conftest INFO Retrieving token for ServiceAccount: mr-test-user-b45f0c11 in namespace mr-rbac-sa
2025-05-07T12:46:40.367240 pyhelper_utils.shell INFO Running oc create token mr-test-user-b45f0c11 -n mr-rbac-sa --duration=10m command
2025-05-07T12:46:41.248802 conftest INFO Successfully retrieved token for SA 'mr-test-user-b45f0c11'
```

Reply:
yeah. if the token was displaying it could have given unauthorized user a window to access the cluster. Such data must be scrubbed from log. Thanks for confirming.

Reply:
It would give access to the MR instance, btw :)

Reply:
For now and any unintended access is bad :)